src/TriggerBinding/SqlTriggerTargetScaler.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    /// <summary>
+    /// Represents the target based scaler returning a target worker count.
+    /// </summary>
+    internal sealed class SqlTriggerTargetScaler : ITargetScaler
+    {
+        private readonly string _userFunctionId;
+        private readonly string _userTableName;
+        private readonly ILogger _logger;
+        private readonly SqlTriggerMetricsProvider _metricsProvider;
+        private readonly int _maxChangesPerWorker;
+
+        public SqlTriggerTargetScaler(string userFunctionId, string userTableName, string connectionString, int maxChangesPerWorker, ILogger logger)
+        {
+            this._userTableName = !string.IsNullOrEmpty(userTableName) ? userTableName : throw new ArgumentNullException(userTableName);
+            this._userFunctionId = !string.IsNullOrEmpty(userFunctionId) ? userFunctionId : throw new ArgumentNullException(userFunctionId);
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._metricsProvider = new SqlTriggerMetricsProvider(connectionString, logger, new SqlObject(userTableName), userFunctionId);
+            this.TargetScalerDescriptor = new TargetScalerDescriptor(userFunctionId);
+            this._maxChangesPerWorker = maxChangesPerWorker;
+        }
+
+        public TargetScalerDescriptor TargetScalerDescriptor { get; }
+
+        public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
+        {
+            SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+
+            return this.GetScaleResultInternal(context, metrics.UnprocessedChangeCount);
+        }
+
+        internal TargetScalerResult GetScaleResultInternal(TargetScalerContext context, long unprocessedChangeCount)
+        {
+            // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
+            int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
```

Comment:
How does InstanceConcurrency map to max changes per worker? The docs don't seem to mention InstanceConcurrency directly and that seems like it would more control the number of workers, not the number of changes per worker.

Reply:
You're correct, docs don't seem to mention it directly. I'll follow up on this.
Cosmos DB extension's TBS does map it to [MaxItemsPerInvocation](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/259962519471b7f64b13c5db4d254f4571167175/src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTargetScaler.cs#L47)